src/GitTfs.VsCommon/TfsHelper.Common.cs

```diff
@@ -201,6 +201,37 @@ public virtual int FindMergeChangesetParent(string path, int targetChangeset, Gi
             });
         }
 
+        public bool TryGetBranchNameBeforeRename(string path, int changeset, out string originalBranchName)
+        {
+            // RecursionType.None sometimes works and sometimes not, so using Full and filtering manually later
+            // The BranchHistoryTreeItem.Relative.IsRequestedItem property seems unreliable so we have to find the entry manually
+            var items = VersionControl.GetBranchHistory([new ItemSpec(path, RecursionType.Full)], new ChangesetVersionSpec(changeset))
```

Comment:
CHECK: Might be better to use `VersionControl.TrackMerges` or some other merge-related method?